### PR TITLE
Format `MatchingNumberOfPrometheusAndCluster` expression

### PR DESCRIPTION
**Pull request by mig4**
to be a bit more readable. No functionality changes.

Towards: https://github.com/giantswarm/giantswarm/issues/15890

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [ ] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: helm/prometheus-meta-operator/templates/prometheus-rules/prometheus-meta-operator.rules.yml [hunk 1]
```diff
@@ -25,7 +25,17 @@ spec:
...
       # This expression list all the cluster IDs that exist and are not being deleted and compares them (using unless) to the running prometheus pods.
       # If a prometheus is missing, this alert will fire. This alert will not check if a prometheus is running when it should not (e.g. deleted cluster)
-      expr: (sum({__name__=~"cluster_service_cluster_info|cluster_operator_cluster_status", status!="Deleting"}) by (cluster_id) unless sum(label_replace(kube_pod_container_status_running{container="prometheus", namespace=~".*-prometheus",namespace!="{{ .Values.Installation.V1.Name }}-prometheus"}, "cluster_id", "$2", "pod", "(prometheus-)(.+)(-.+)") ) by (cluster_id)) > 0
+      expr: |
+        (
+          sum by(cluster_id) (
+            {__name__=~"cluster_service_cluster_info|cluster_operator_cluster_status",status!="Deleting"}
+          ) unless sum by(cluster_id) (
+            label_replace(
+              kube_pod_container_status_running{container="prometheus",namespace!="{{ .Values.Installation.V1.Name }}-prometheus",namespace=~".*-prometheus"},
```
Comment (JosephSalisbury) [inline helm/prometheus-meta-operator/templates/prometheus-rules/prometheus-meta-operator.rules.yml:34]:
```suggestion
              kube_pod_container_status_running{container="prometheus", namespace!="{{ .Values.Installation.V1.Name }}-prometheus", namespace=~".*-prometheus"},
```

File: helm/prometheus-meta-operator/templates/prometheus-rules/prometheus-meta-operator.rules.yml [hunk 2]
```diff
@@ -25,7 +25,17 @@ spec:
         description: This alert is used to ensure we have as many workload cluster prometheus as we have workload cluster CR.
       # This expression list all the cluster IDs that exist and are not being deleted and compares them (using unless) to the running prometheus pods.
       # If a prometheus is missing, this alert will fire. This alert will not check if a prometheus is running when it should not (e.g. deleted cluster)
-      expr: (sum({__name__=~"cluster_service_cluster_info|cluster_operator_cluster_status", status!="Deleting"}) by (cluster_id) unless sum(label_replace(kube_pod_container_status_running{container="prometheus", namespace=~".*-prometheus",namespace!="{{ .Values.Installation.V1.Name }}-prometheus"}, "cluster_id", "$2", "pod", "(prometheus-)(.+)(-.+)") ) by (cluster_id)) > 0
+      expr: |
+        (
+          sum by(cluster_id) (
+            {__name__=~"cluster_service_cluster_info|cluster_operator_cluster_status",status!="Deleting"}
```
Comment (JosephSalisbury) [inline helm/prometheus-meta-operator/templates/prometheus-rules/prometheus-meta-operator.rules.yml:31]:
```suggestion
            {__name__=~"cluster_service_cluster_info|cluster_operator_cluster_status", status!="Deleting"}
```

**Review by JosephSalisbury — APPROVED** [review]
personal preference, but i prefer having spaces between label selectors in promql expressions

**Comment by mig4** [comment]
> personal preference, but i prefer having spaces between label selectors in promql expressions

Good catch, me too, just didn't notice this as I copied from Prometheus.

**Pull request merged**